### PR TITLE
Document Popular Plugins and Archive Old Plugins 

### LINKED ISSUE #791: Document Plugins
### What should we change and why?

Building on #785, document how to enable plugins from the CLI and GUI

Additionally, provide brief examples of how to use the most popular plugins:
* ixbrl-viewer
* inlineXbrlDocumentSet
* validate/ESEF_2022
* validate/EFM
* EdgarRenderer

### PR DESCRIPTION
**Pull request by austinmatherne-wk**
#### Reason for change
resolves #791

* Add documentation for how to use many of the most commonly used plugins.
* Many of the plugins are broken/untested. Move many of them to an archive directory that can be used from the plugin manager browse dialog instead of cluttering the built-in menu.
* Update Windows installer/uninstaller to clear install directory so archived plugins aren't left behind (this also addresses issues with legacy installs where old python libs are still present).

#### Steps to Test
* Check the new RTD [build for this branch](https://arelle.readthedocs.io/en/xt-3104/plugins/popular/popular_plugins.html)
* With the latest Arelle build enable a plugin that's archived in this PR.
* Update to this branch.
* Launch Arelle and confirm that a warning is logged informing the user that the plugin could not be located and may have been archived.
* Verify that the plugin is no longer enabled.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by derekgengenbacher-wf — APPROVED** [review]
Looks great!!

**Review by aaroncameron-wk — DISMISSED** [review]

File: arelle/plugin/validate/ESEF/__init__.py [hunk 1]
```diff
@@ -364,6 +360,12 @@ def modelTestcaseVariationReportPackageIxdsOptions(
 __pluginInfo__ = {
     # Do not use _( ) in pluginInfo itself (it is applied later, after loading
     "name": "Validate ESMA ESEF",
+    "aliases": [
+        # Aliases for when ESEF validation was handled by multiple plugins.
+        "Validate ESMA ESEF-2022",
+        "validate/ESEF_2022",
+        "validate\ESEF_2022",
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/ESEF/__init__.py:367]: Fine to leave this, but aliases (and the search terms compared against them) do get normalized to replace `\` with `/` so these will evaluate to the same alias.

**Review by aaroncameron-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Pull request merged**